.github: Clone pytorch to separate directory

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#61932 .github: Clone pytorch to separate directory**

Clones pytorch to a separate directory for each run so that they do not
overlap with each other

To avoid errors like 
![image](https://user-images.githubusercontent.com/1700823/126393228-ef70b103-0312-497a-9b3f-a5046b88d537.png)


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29801875](https://our.internmc.facebook.com/intern/diff/D29801875)

seemethere committed:
.github: Explicitly clean GITHUB_WORKSPACE

Explicitly removes everything in GITHUB_WORKSPACE for windows workflows

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801875).

seemethere committed:
Update on ".github: Explicitly clean GITHUB_WORKSPACE"


Explicitly removes everything in GITHUB_WORKSPACE for windows workflows

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29801875](https://our.internmc.facebook.com/intern/diff/D29801875)

[ghstack-poisoned]

seemethere committed:
Update on ".github: Clone pytorch to separate directory"


Clones pytorch to a separate directory for each run so that they do not
overlap with each other

To avoid errors like 
![image](https://user-images.githubusercontent.com/1700823/126393228-ef70b103-0312-497a-9b3f-a5046b88d537.png)


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29801875](https://our.internmc.facebook.com/intern/diff/D29801875)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801875).

zhouzhuojie commented on .github/templates/windows_ci_workflow.yml.j2:
```diff
@@ -34,24 +34,25 @@ env:
     runs-on: "windows.4xlarge"
+    defaults:
+      run:
+        working-directory: pytorch-${{ github.run_id }}
```
even with its new working-directory, let's also add the cleanup script. Understandable that scale-down can do some cleanup, but there's a small chance that the runner can run for a long time and can run out of disk

seemethere replied:
Fair enough, I'll add that in

seemethere committed:
Update on ".github: Clone pytorch to separate directory"


Clones pytorch to a separate directory for each run so that they do not
overlap with each other

To avoid errors like 
![image](https://user-images.githubusercontent.com/1700823/126393228-ef70b103-0312-497a-9b3f-a5046b88d537.png)


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29801875](https://our.internmc.facebook.com/intern/diff/D29801875)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801875).

seemethere committed:
Update on ".github: Clone pytorch to separate directory"


Clones pytorch to a separate directory for each run so that they do not
overlap with each other

To avoid errors like 
![image](https://user-images.githubusercontent.com/1700823/126393228-ef70b103-0312-497a-9b3f-a5046b88d537.png)


Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D29801875](https://our.internmc.facebook.com/intern/diff/D29801875)

[ghstack-poisoned]

seemethere commented:
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29801875).
